### ISSUE TITLE
Re-generate citations using the new pipeline for references

### ISSUE THREAD

**Issue by bruvellu** [nelas/cifonauta#333]
The citations for existing references were generated using a different format. The new references are being imported from crossref with the APA citation style. For references with DOI this could be re-generated, for the others, we will need to manually edit them.

**Comment by castilhovirginia** [comment]
I'm reviewing references via admin. Locating the DOI, placing it in the name and DOI field, and modifying the citation field according to the DOI API.

Slug is necessary? ID is the link with the media. If it´s necessary, I think it should to run some routine to exchange the slug value for the DOI value. 

I put "N-DOI", followed by the name field of the first author and the publication date, in name and DOI field references that do not have a DOI.

**Comment by bruvellu** [comment]
Using DOI in the URL can be a problem because some characters allowed in DOIs need to be encoded, like `%23` for `#`.

Instead, I would use a bibkey-like as name and slug. For example, this current reference:

```
Name: 10.1371/journal.pone.0002429
Slug: 3878021152
Citation: Lindner, A., Cairns, S. D., & Cunningham, C. W. (2008). From Offshore to Onshore: Multiple Origins of Shallow-Water Corals from Deep-Sea Ancestors. PLoS ONE, 3(6), e2429. https://doi.org/10.1371/journal.pone.0002429
DOI: 10.1371/journal.pone.0002429
```

Can become:

```
Name: Lindner2008a
Slug: lindner2008a
Citation: Lindner, A., Cairns, S. D., & Cunningham, C. W. (2008). From Offshore to Onshore: Multiple Origins of Shallow-Water Corals from Deep-Sea Ancestors. PLoS ONE, 3(6), e2429. https://doi.org/10.1371/journal.pone.0002429
DOI: 10.1371/journal.pone.0002429
```

This way, `Lindner2008a` can be used as a readable title for the reference page.

I would also keep the DOI field empty if there's no DOI, that's easier to check for adding the link in the templates.

**Comment by bruvellu** [comment]
Hi, I updated the reference import to use the DOI field (not name) for checking for existing references (411c1e3330541f54571d23bcad2260836da1255a), and to generate a bibkey with last name + pub date + two random letters (example: `Vellutini2017-nt`) and save it as the reference name (846c276509220471a2e6ac8871b303d21530cf82).

**Comment by bruvellu** [comment]
That's how the reference page looks like with the new system:

**URL:** `/reference/vieira2012-ab/`

![Image](https://github.com/user-attachments/assets/1d6acbad-09bb-4c67-8ae9-6d2dde5a48f9)

**Comment by bruvellu** [comment]
Hi, with 35de1a434fde709ccfc36c793736f0a4c2dbfedf, I finished improving the references. Here are the [updates](https://github.com/bruvellu/cifonauta/compare/d6d7d897afa207d565eaa4ba415adc4a82271876..35de1a434fde709ccfc36c793736f0a4c2dbfedf):

- Improve detection of valid DOIs and author names
- Generate reference name on import and from citation
- Standardize references to NameYear-xy and citations to APA style
- Fix several display bugs and improved styles
- Update Reference admin to show additional fields

Everything is on staging. Ready for review!

**Comment by castilhovirginia** [comment]
I tested including a reference when editing data and there is a problem. There is no error message when inserting duplicate references and the registration is done (see reference-ids 158 and 159 that I inserted). Can you check this?

**Comment by bruvellu** [comment]
Thanks for noticing. Duplicate detection fixed in 50d2dfe89b340d703a005411e671e668d6f9137c

**Comment by bruvellu** [comment]
Add DOI to display name for enabling search a327206f20c55a1bc6dd0136e23ca80e5b62a1ab
